eclair-core/src/main/scala/fr/acinq/eclair/channel/Register.scala

```diff
@@ -26,25 +27,31 @@ import fr.acinq.eclair.channel.Register._
  * Created by PM on 26/01/2016.
  */
 
-class Register extends Actor with ActorLogging {
+class Register() extends Actor with ActorLogging {
 
   context.system.eventStream.subscribe(self, classOf[ChannelCreated])
   context.system.eventStream.subscribe(self, classOf[AbstractChannelRestored])
   context.system.eventStream.subscribe(self, classOf[ChannelIdAssigned])
   context.system.eventStream.subscribe(self, classOf[ShortChannelIdAssigned])
 
+  // @formatter:off
+  private case class ChannelTerminated(channel: ActorRef, channelId_opt: Option[ByteVector32], shortChannelId_opt: Option[ShortChannelId])
```

Comment:
Based on the current code, we can make `channelId` non-optional:
```suggestion
  private case class ChannelTerminated(channel: ActorRef, channelId: ByteVector32, shortChannelId_opt: Option[ShortChannelId])
```

Is there a feature branch reason for making it optional?

Also, `shortChannelId_opt` is never set...?